osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyCatchComboCounter.cs

```diff
@@ -1,64 +1,112 @@
 
+        public bool UsesFixedAnchor { get; set; }
+
+        private int lastDisplayedCombo;
+
+        bool ISerialisableDrawable.SupportsClosestAnchor => false;
+
         public LegacyCatchComboCounter()
         {
             AutoSizeAxes = Axes.Both;
 
+            AlwaysPresent = true;
             Alpha = 0f;
-            Anchor = Anchor.Centre;
-            Origin = Anchor.Centre;
-            Scale = new Vector2(0.8f);
 
-            InternalChildren = new Drawable[]
+            InternalChild = new UprightAspectMaintainingContainer
             {
-                explosion = new LegacyRollingCounter(LegacyFont.Combo)
-                {
-                    Alpha = 0.65f,
-                    Blending = BlendingParameters.Additive,
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    Scale = new Vector2(1.5f),
-                },
-                counter = new LegacyRollingCounter(LegacyFont.Combo)
+                AutoSizeAxes = Axes.Both,
+                Children = new[]
                 {
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                },
+                    explosion = new LegacyRollingCounter(LegacyFont.Combo)
+                    {
+                        Alpha = 0.65f,
+                        Blending = BlendingParameters.Additive,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(1.5f),
+                    },
+                    counter = new LegacyRollingCounter(LegacyFont.Combo)
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                    },
+                }
             };
+
+            UsesFixedAnchor = true;
         }
 
-        private int lastDisplayedCombo;
+        [Resolved]
+        private GameplayState gameplayState { get; set; } = null!;
+
+        [Resolved]
+        private DrawableRuleset? drawableRuleset { get; set; }
+
+        [Resolved]
+        private ISkinSource skin { get; set; } = null!;
+
+        private IBindable<JudgementResult> lastJudgementResult = null!;
+        private IBindable<int> combo = null!;
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            combo = gameplayState.ScoreProcessor.Combo.GetBoundCopy();
+            lastJudgementResult = gameplayState.LastJudgementResult.GetBoundCopy();
+            lastJudgementResult.BindValueChanged(onNewJudgement, true);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (drawableRuleset != null)
+            {
+                var catchPlayfield = (CatchPlayfield)drawableRuleset.Playfield;
+                X = Parent.AsNonNull().ToLocalSpace(catchPlayfield.Catcher.ScreenSpaceDrawQuad.Centre).X;
```

Comment:
I'm not 100% sure about this performance-wise, but it looks to work seamlessly.

Reply:
Probably okay. Minor refactor applied for legibility.
;